torch/csrc/autograd/profiler.cpp

```diff
@@ -569,6 +648,44 @@ thread_event_lists disableProfiler(c10::optional<ProfilerDisableOptions> profile
     at::removeCallback(state_ptr->callbackHandle());
   }
 
+#ifdef USE_KINETO
+  if (state_ptr->config().state == ProfilerState::KINETO) {
+    auto k_events = libkineto::api().stopTrace();
+    std::unordered_map<uint32_t, std::unordered_map<uint32_t, std::vector<Event>>> events;
+    for (auto& k_evt : k_events) {
+      auto& evt_list = events[k_evt.deviceId][k_evt.threadId];
+      Event push_evt(
+          EventKind::PushRange,
+          at::StringView(k_evt.name),
+          k_evt.threadId,
+          false,
+          k_evt.correlationId);
+      push_evt.setDevice(k_evt.deviceId);
```

Comment:
Hmm... maybe we can add an adapter instead to avoid copying? Or do we want to copy anyway? Currently I keep the profiler "busy" until reset() is called, and the events returned is alive until then. We could add clone or similar if we need to extend the lifetime.